src/test/lib/storage/constraints/foreign_key_constraint_test.cpp

```diff
@@ -0,0 +1,200 @@
+#include <numeric>
+
+#include "base_test.hpp"
+#include "storage/constraints/foreign_key_constraint.hpp"
+#include "storage/table.hpp"
+
+namespace hyrise {
+
+class ForeignKeyConstraintTest : public BaseTest {
+ protected:
+  void SetUp() override {
+    _table_a = Table::create_dummy_table({{"a", DataType::Int, false},
+                                          {"b", DataType::Int, false},
+                                          {"c", DataType::Int, false},
+                                          {"d", DataType::Int, false},
+                                          {"e", DataType::Int, false}});
+    _table_b = Table::create_dummy_table({{"x", DataType::Int, false}, {"y", DataType::Int, false}});
+    _table_c = Table::create_dummy_table({{"u", DataType::Int, false}, {"v", DataType::Int, false}});
+  }
+
+  std::shared_ptr<Table> _table_a, _table_b, _table_c;
+};
+
+TEST_F(ForeignKeyConstraintTest, OrderedColumnIDs) {
+  // To handle equivalent foreign key constraints / INDs with swapped columns, we sort the foreign key columns and apply
+  // the permutation to the primary key columns.
+  const auto foreign_key_constraint =
+      ForeignKeyConstraint{{ColumnID{1}, ColumnID{0}}, _table_b, {ColumnID{3}, ColumnID{4}}, _table_a};
+
+  EXPECT_EQ(foreign_key_constraint.foreign_key_columns().size(), 2);
+  EXPECT_EQ(foreign_key_constraint.foreign_key_columns().front(), ColumnID{0});
+  EXPECT_EQ(foreign_key_constraint.foreign_key_columns().back(), ColumnID{1});
+
+  EXPECT_EQ(foreign_key_constraint.primary_key_columns().size(), 2);
+  EXPECT_EQ(foreign_key_constraint.primary_key_columns().front(), ColumnID{4});
+  EXPECT_EQ(foreign_key_constraint.primary_key_columns().back(), ColumnID{3});
+
+  EXPECT_EQ(foreign_key_constraint.foreign_key_table(), _table_b);
+  EXPECT_EQ(foreign_key_constraint.primary_key_table(), _table_a);
+
+  // Try a larger one.
+  auto column_ids = std::vector<ColumnID>(100);
+  std::iota(column_ids.begin(), column_ids.end(), ColumnID{0});
+  auto reverse_column_ids = column_ids;
+  std::sort(reverse_column_ids.begin(), reverse_column_ids.end(), std::greater<ColumnID>{});
+  const auto large_foreign_key_constraint =
+      ForeignKeyConstraint{reverse_column_ids, _table_a, reverse_column_ids, _table_b};
+  EXPECT_EQ(large_foreign_key_constraint.foreign_key_columns(), column_ids);
+  EXPECT_EQ(large_foreign_key_constraint.primary_key_columns(), column_ids);
+}
+
+TEST_F(ForeignKeyConstraintTest, AddForeignKeyConstraints) {
+  EXPECT_EQ(_table_a->soft_foreign_key_constraints().size(), 0);
+
+  const auto foreign_key_constraint_1 = ForeignKeyConstraint{{ColumnID{0}}, _table_a, {ColumnID{0}}, _table_b};
```

Comment:
These repetitions for each FK don't look nice. Can we get rid of them?

Reply:
```c++
const auto foreign_key_constraint_1 = ForeignKeyConstraint{{ColumnID{0}}, _table_a, {ColumnID{0}}, _table_b};
_table_a->add_soft_foreign_key_constraint(foreign_key_constraint);
```
?

Also like this better, but adapted to the style used in `table_key_constraint_test.hpp`